backend/server/routers/programs.py

```diff
@@ -136,11 +136,12 @@ def getMinors(programCode: str):
     return {"minors": minrs}
 
 def convertSubgroupObjectToCoursesDict(object: str, description: str|list[str]) -> dict[str, str]:
-    """ gets a subgroup object (format layed out in the processor) and fetches the exact courses its referring to """
+    """ gets a subgroup object (format laid out in the processor) and fetches the exact courses its referring to """
     if " or " in object:
         return {c: description[index] for index, c in enumerate(object.split(" or "))}
     elif not re.match(r"[A-Z]{4}[0-9]{4}", object):
-        return search(object)
+        # TODO(josh): ensure this is accurate
```

Comment:
@Peedee2002 when does this actually get used? I can't quite figure out the intended purpose of this

Reply:
its an artifact of how we store program data. Some strings show up like:
"COMP9" - this indicates any level 9 course
"COMP3121 or COMP3821" - this means that we can pick one of two. 

Maybe the correct solution is to push this logic to the data section actually.

Reply:
Yeah that might be a good idea but sounds like I should bring back the regex based search for this case

Reply:
> Yeah that might be a good idea but sounds like I should bring back the regex based search for this case in the meantime

sounds good. We should make it unavailable to the frontend tho.

Reply:
If you mean not providing an endpoint to the function, this is already the case

Reply:
Are there any cases where this would require matching on the title of the course or is it only ever the course codes?

Reply:
only codes.